### DISCUSSION TITLE
Weekly Stand-Up Discussion:  Sprint #3

### DISCUSSION THREAD

**Discussion by Rankin-Craig** [JamesSG2/Handwriting-Conversion-Application#14] [General]
A Stand-Up Discussion or meeting is something that would take place on a daily basis in a real-world software design team setting. It is called a "Stand-Up meeting" because it is meant to be done very quickly to inform your other group members about the work you have been doing. Since we are attempting to mimic this situation we will complete a Stand-Up discussion every week.

Please answer the following questions regarding the work you have done from Friday 4/2 until Wednesday 4/7:

What specific tasks did you work on during this time period (Sprint #3)? Explain each one.
What task did you spend the most time on?
Did you run into any problems or roadblocks that you may need to discuss with the group or others? - Describe those issues.
Evaluate the responses in this discussion board from others. Respond by offering your help or advice to someone else.

**Comment by rover-t** [comment]
1. I've worked on creating a function that receives a string to write and output and image of text in your handwriting. I also corrected an error that was causing capital and lower case letters to be stored in the same location.
2. I spent the most time working on the new function.
3. I think it might be a good idea to switch from storing files by naming them after the character e.g. "a_0" to using their Unicode number e.g. "97_0". This would prevent some issues with special characters and capitalization.

**Comment by JosephHaugen** [comment]
1. learned about potential GUI libraries for python, I think the built in GUI tools would be ideal
2. designed basic UI with button that can select files from a windows machine.

**Comment by BigWub67** [comment]
1. I started learning tkinter and making the start screen to the app. Right now it's just two buttons that don't do anything, but they do scale to window size (not in real time though)
2. I think I spent the most time learning tkinter.
3. I haven't run into anything yet.

**Comment by JamesSG2** [comment]
1.   I updated my wip-confidence-levels branch with the newest code from the dev branch, and began running several tests with the current data and some of my own. I spent most of my time researching how implement a confidence level method/class from tesseract, and trying to get it to return a confidence level in our code, for each individual charactor and/or the bounding boxes (-- the bounding boxes though appear to be created by OpenCV not Tessteract, this requires a seperate cv2 method). I begun writing some preliminary code for returning individual character confidence, and getting it to reject low confidence characters. I'll post it to my branch later this week when it's fully funtional. Also, later on we'll probably have to have the user enter/handwrite several versions of the same pangram to reduce the likelyhood that the same charater is rejected everytime (and thus doesn't make it to the output).

2. As stated above: I spent most of my time researching how implement a confidence level method/class from tesseract, and trying to get it to return a confidence level in our code.

3. It's a bit unclear where in our code the predictions are being made for each character, which I'll need to implement my confidence level method which stores a confidence level for each character. I'll talk to Trevor about this in the meeting tomorrow since he wrote the code.

4. After running some tests with longer essays/paragraphs I noticed that the output doesn't yet return after each line, and the size/window limits cut-off the rest of the writing. We should probablly fix that in order for the program to be practically useful. I'll see what I can do.
